### DISCUSSION TITLE
[Nuxt 3] How to use Pinia outside of a component?

### DISCUSSION THREAD

**Discussion by onurusluca** [vuejs/pinia#2379] [Help and Questions] [answered]
I am trying to use Pinia with Phaser but I can't get it working.
It says: `"getActivePinia()" was called but there was no active Pinia.`

So I read the docs but didn't understand. How do I use Pinia in a .ts file outside of the reach of Nuxt?
https://pinia.vuejs.org/ssr/nuxt.html#Using-the-store-outside-of-setup-

This problem seems to be not fixed:
https://github.com/vuejs/pinia/discussions/2245
https://github.com/vuejs/pinia/discussions/1212

**Comment by posva** [comment, marked as answer]
Ensure `app.use(pinia)` is called before any `useStore()` function is called or manually call `setActivePinia(pinia)` before but you might encounter missing things like plugins. Otherwise, refer to this [guide](https://pinia.vuejs.org/core-concepts/outside-component-usage.html#Single-Page-Applications)

**Reply by onurusluca** [reply]
Thank you for the reply.
In a .ts file outside of Nuxt scope, let's say I want to do that.

```
import { setActivePinia, createPinia } from "pinia";

const pinia = createPinia();
setActivePinia(pinia);
const mainStore = useMainStore();
```

It seems to be working. Is this correct way of doing this? If so, I would like to add this to docs cause it is difficult to understand this from what is written there.

**Reply by posva** [reply]
You might encounter some missing things. The right way is to ensure pinia is installed before calling useStore. That’s why it’s not in docs

**Reply by onurusluca** [reply]
The problem is that with Nuxt 3 we have no main.ts file so we can't really decide when to initiate Pinia. [Module ](https://pinia.vuejs.org/ssr/nuxt.html) handle that. Wish there was another way.

**Reply by posva** [reply]
I think you can if you add the plugin with a module: https://nuxt.com/docs/api/advanced/kit#plugins. But I thought pinia plugin would be added before user plugins. It might be a design choice of Nuxt.
